packages/iov-core/src/signingservercore.spec.ts

```diff
@@ -196,6 +196,37 @@ describe("SigningServerCore", () => {
 
       core.shutdown();
     });
+
+    it("handles exceptions in callback", async () => {
+      const profile = new UserProfile();
+      const wallet = profile.addWallet(
+        Ed25519HdWallet.fromMnemonic(
+          "option diagram plastic million educate they arrow fat comic excite abandon green",
+        ),
+      );
+
+      await profile.createIdentity(wallet.id, defaultChainId, HdPaths.iov(0));
+      await profile.createIdentity(wallet.id, defaultChainId, HdPaths.iov(1));
+      await profile.createIdentity(wallet.id, defaultChainId, HdPaths.iov(2));
+      await profile.createIdentity(wallet.id, defaultChainId, HdPaths.iov(3));
+
+      async function throwingCallback(
+        _1: string,
+        _2: ReadonlyArray<PublicIdentity>,
+      ): Promise<ReadonlyArray<PublicIdentity>> {
+        throw new Error("Something broken in here!");
+      }
+
+      const signer = new MultiChainSigner(profile);
+      const core = new SigningServerCore(profile, signer, throwingCallback, defaultSignAndPostCallback);
+
+      await core
+        .getIdentities("Login to XY service", [defaultChainId])
+        .then(() => fail("must not resolve"))
+        .catch(error => expect(error).toMatch(/internal server error/i));
```

Comment:
So we guarantee an error that does not leak internal info of the server. This is good (we also had a long discussion of redacting errors in weave), but can make it hard to debug. It would be good if the server (separate project #400 right?) logs any errors, so it is possible at least for the server code to report them somehow, if we don't share them with the untrusted client.

Reply:
I think the signing server (i.e. the browser extension) should log those errors. That way we have a clear separation of internal and external and a developer can always check the browser extension's logs for debugging.